src-gui/package.json

```diff
@@ -4,6 +4,8 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
+    "check-bindings": "typeshare --lang=typescript --output-file __temp_bindings.ts ../swap/src && dprint fmt __temp_bindings.ts && diff -wbB __temp_bindings.ts ./src/models/tauriModel.ts && rm __temp_bindings.ts",
+    "gen-bindings": "typeshare --lang=typescript --output-file ./src/models/tauriModel.ts ../swap/src && dprint fmt ./src/models/tauriModel.ts",
```

Comment:
I think we still need to execute this before building for production. Vite will probably only run the watch config in preview mode.

You can add a prebuid script that executes gen-bindings.

See https://docs.npmjs.com/cli/v10/using-npm/scripts#pre--post-scripts

Reply:
That might be, but it might also just be that it  works

Reply:
I tested it and `vite build` does generate the bindings, however `tsc` is invoked before `vite build` which might lead to a compile error... I'll add a call to `gen-bindings` even before `tsc`.